main.py

```diff
@@ -5,9 +5,19 @@
 import json
 
 data_list = []
-counter = 0
+parsingCounter = 0
 errorCounter = 0
 
+# --------- Loading Animation/Message ---------
+loadingCounter = 0
+animation = "|/-\\"
```

Comment:
I'm not sure about making that variable global but it made more sense than creating it each time that function was called.